### PR TITLE
Modified controller to get receipts by day, week, month, and year

### PR DESCRIPTION
**Pull request by isabelt220**
Get all receipts by:
- [x] day
- [x] week
- [x] month
- [x] year
- [x] user determined start and end dates

### CODE REVIEW

**Comment by isabelt220** [comment]
I feel like there's a lot of duplicate code in get_all_receipts() but I haven't found a way to reduce that code yet. Any feedback would be appreciated!

File: server/api/receipt_controller.py [hunk 1]
```diff
@@ -34,23 +34,50 @@ def create():
...
+    elif year is not None and month is not None and date is None:
         num_days = calendar.monthrange(year, month)[1]
         start_date = datetime.date(year, month, 1)
         end_date = datetime.date(year, month, num_days)
 
+        receipts = (Receipt.query
+                    .filter(Receipt.receipt_date <= end_date)
+                    .filter(Receipt.receipt_date >= start_date)
+                    .order_by(Receipt.receipt_date.desc()))
+    elif year is not None and month is not None and date is not None and weekly is False:
```
Comment (bonnieli) [inline server/api/receipt_controller.py]:
```
elif year is not None and month is not None and date is not None:
  if weekly:
    d = datetime.date(year, month, date)
    receipts = (Receipt.query
                    .filter(Receipt.receipt_date == d)
                    .order_by(Receipt.receipt_date.desc()))
  else:
    start_date = datetime.date(year, month, date)
        day_of_week = start_date.isoweekday()
        if day_of_week != 7:
            start_date -= datetime.timedelta(days=day_of_week)
        end_date = start_date + datetime.timedelta(days=6)

        receipts = (Receipt.query etc etc
```

Comment (bonnieli) [reply server/api/receipt_controller.py]: this just reduces logic repeated

File: server/api/receipt_controller.py [hunk 2]
```diff
@@ -34,23 +34,50 @@ def create():
...
+                    .filter(Receipt.receipt_date <= end_date)
+                    .filter(Receipt.receipt_date >= start_date)
+                    .order_by(Receipt.receipt_date.desc()))
+    elif year is not None and month is not None and date is None:
         num_days = calendar.monthrange(year, month)[1]
         start_date = datetime.date(year, month, 1)
         end_date = datetime.date(year, month, num_days)
 
+        receipts = (Receipt.query
+                    .filter(Receipt.receipt_date <= end_date)
```
Comment (bonnieli) [inline server/api/receipt_controller.py]: a lot of this same filter 

```
receipts = (Receipt.query
                    .filter(Receipt.receipt_date <= end_date)
                    .filter(Receipt.receipt_date >= start_date)
                    .order_by(Receipt.receipt_date.desc()))
```

is the same. to reduce the # of lines, you can perhaps look into creating a helper func

Comment (isabelt220) [reply server/api/receipt_controller.py]: I realized that pretty much all of the receipt queries could be formatted into the same filter so I made some changes to the code in some of the IF blocks so all of them can use the same filter.

**Review by ethanlow23 — APPROVED** [review]

**Pull request closed without merging**